src/CodeGen_LLVM.cpp

```diff
@@ -603,6 +608,12 @@ void CodeGen_LLVM::begin_func(LoweredFunc::LinkageType linkage, const std::strin
     BasicBlock *block = BasicBlock::Create(*context, "entry", function);
     builder->SetInsertPoint(block);
 
+    // Put the global buffers in the symbol table.
+    for (const auto &buf : input_module->buffers()) {
+        llvm::Value *buf_ptr = sym_get(buf.name() + ".buffer");
```

Comment:
We could get rid of the global special case stuff in push/pop buffer by changing the name of the global buffer to be just buf.name(). Then push_buffer would just unconditionally push buf_ptr->getName() + ".buffer". It also might actually be a bug not to do this, because push_buffer pushes buf_ptr->getName() to "external_buffers", I'm not sure if global buffers meet the alignment requirements we want or not.